interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp

```diff
@@ -416,47 +500,152 @@ namespace cling {
+          std::string RPath;
+          std::string RunPath;
+          std::vector<StringRef> Deps;
+          const std::string NewLibP_str = NewLibPath->GetFullName();
+          llvm::StringRef NewLibP(NewLibP_str);
+          auto ObjFileOrErr =
+            llvm::object::ObjectFile::createObjectFile(NewLibP);
+          if (llvm::Error Err = ObjFileOrErr.takeError()) {
+            if (DEBUG > 1) {
+              std::string Message;
+              handleAllErrors(std::move(Err), [&](llvm::ErrorInfoBase &EIB) {
+                Message += EIB.message() + "; ";
+              });
+              cling::errs()
+                << "Dyld::ScanForLibraries: Failed to read object file "
+                << NewLibP.str() << " Errors: " << Message << "\n";
+            }
+            return nullptr;
+          }
+          llvm::object::ObjectFile *BinObjF = ObjFileOrErr.get().getBinary();
+          if (BinObjF->isELF()) {
+            if (const auto* ELF = dyn_cast<ELF32LEObjectFile>(BinObjF))
+              HandleDynTab(ELF->getELFFile(), FileName, RPath, RunPath, Deps);
+            else if (const auto* ELF = dyn_cast<ELF32BEObjectFile>(BinObjF))
+              HandleDynTab(ELF->getELFFile(), FileName, RPath, RunPath, Deps);
+            else if (const auto* ELF = dyn_cast<ELF64LEObjectFile>(BinObjF))
+              HandleDynTab(ELF->getELFFile(), FileName, RPath, RunPath, Deps);
+            else if (const auto* ELF = dyn_cast<ELF64BEObjectFile>(BinObjF))
+              HandleDynTab(ELF->getELFFile(), FileName, RPath, RunPath, Deps);
+
+          } else if (BinObjF->isMachO()) {
+            MachOObjectFile *Obj = (MachOObjectFile*)BinObjF;
+            for (const auto &Command : Obj->load_commands()) {
+              if (Command.C.cmd == MachO::LC_LOAD_DYLIB) {
+                  //Command.C.cmd == MachO::LC_ID_DYLIB ||
+                  //Command.C.cmd == MachO::LC_LOAD_WEAK_DYLIB ||
+                  //Command.C.cmd == MachO::LC_REEXPORT_DYLIB ||
+                  //Command.C.cmd == MachO::LC_LAZY_LOAD_DYLIB ||
+                  //Command.C.cmd == MachO::LC_LOAD_UPWARD_DYLIB ||
+                MachO::dylib_command dylibCmd =
+                  Obj->getDylibIDLoadCommand(Command);
+                Deps.push_back(StringRef(Command.Ptr + dylibCmd.dylib.name));
+              }
+              else if (Command.C.cmd == MachO::LC_RPATH) {
+                MachO::rpath_command rpathCmd = Obj->getRpathCommand(Command);
+                CombinePaths(RPath, Command.Ptr + rpathCmd.path);
+              }
+            }
+          }
+
+          for (StringRef dep : Deps) {
+            std::string dep_full =
+              m_DynamicLibraryManager.lookupLibrary(dep, RPath, RunPath, NewLibP, false);
+            const LibraryPath* DepLibPath = HandleLib(dep_full, level + 1);
+            //NewLibPath. add dependency lib to dep list?
+            if (!DepLibPath) {
+              const_cast<LibraryPath*>(NewLibPath)->m_LibDeps.push_back(DepLibPath);
```

Comment:
Why do you only `push_back` `nullptr`s?  Is the `if` condition meant to be inverted? This doesn't seem to be covered by a test?